dll/cpl/input/settings_page.c

```diff
@@ -11,192 +11,343 @@
+        insert.hInsertAfter = TVI_LAST;
+        insert.item         = item;
+        hItem = TreeView_InsertItem(hwndList, &insert);
+
+        // Keyboard
+        ZeroMemory(&item, sizeof(item));
+        item.mask           = TVIF_TEXT | TVIF_IMAGE | TVIF_PARAM | TVIF_SELECTEDIMAGE | TVIF_STATE;
+        item.pszText        = szKeyboard;
+        item.iImage         = KeyboardImageIndex;
+        item.iSelectedImage = KeyboardImageIndex;
+        item.lParam         = 0xFFFF;
+        if (bBold)
+        {
+            item.state = item.stateMask = TVIS_BOLD;
+        }
+        insert.hParent      = hItem;
+        insert.hInsertAfter = TVI_LAST;
+        insert.item         = item;
+        hItem = TreeView_InsertItem(hwndList, &insert);
+    }
+    else
     {
-        HICON hLayoutIcon;
+        // Language
+        ZeroMemory(&item, sizeof(item));
+        item.mask           = TVIF_STATE | TVIF_HANDLE;
+        item.hItem          = hItem;
+        item.stateMask      = TVIS_BOLD;
+        if (TreeView_GetItem(hwndList, &item) && bBold && !(item.state & TVIS_BOLD))
+        {
+            item.mask = TVIF_STATE | TVIF_HANDLE;
+            item.hItem = hItem;
+            item.state = item.stateMask = TVIS_BOLD;
+            TreeView_SetItem(hwndList, &item);
+        }
 
-        hLayoutIcon = CreateLayoutIcon(pInputNode->pszIndicator,
-                                       (pInputNode->wFlags & INPUT_LIST_NODE_FLAG_DEFAULT));
-        if (hLayoutIcon != NULL)
+        // Keyboard
+        hItem = TreeView_GetChild(hwndList, hItem);
+        ZeroMemory(&item, sizeof(item));
+        item.mask           = TVIF_STATE | TVIF_HANDLE;
+        item.hItem          = hItem;
+        item.stateMask      = TVIS_BOLD;
+        if (TreeView_GetItem(hwndList, &item) && bBold && !(item.state & TVIS_BOLD))
         {
-            ImageIndex = ImageList_AddIcon(hImageList, hLayoutIcon);
-            DestroyIcon(hLayoutIcon);
+            item.mask = TVIF_STATE | TVIF_HANDLE;
+            item.hItem = hItem;
+            item.state = item.stateMask = TVIS_BOLD;
+            TreeView_SetItem(hwndList, &item);
         }
     }
 
-    ZeroMemory(&item, sizeof(item));
-    item.mask    = LVIF_TEXT | LVIF_PARAM | LVIF_IMAGE;
-    item.pszText = pInputNode->pLocale->pszName;
-    item.iItem   = ListView_GetItemCount(hwndList);
-    item.lParam  = (LPARAM)pInputNode;
-    item.iImage  = ImageIndex;
-    ItemIndex = ListView_InsertItem(hwndList, &item);
+    // Input method
+    if (hItem)
+    {
+        // FIXME: Load 1st resource icon from "IME File"...
+        hInputMethodIcon = (HICON)LoadImageW(hApplet,
+                                             MAKEINTRESOURCEW(IDI_DOT),
+                                             IMAGE_ICON,
+                                             GetSystemMetrics(SM_CXSMICON),
+                                             GetSystemMetrics(SM_CYSMICON),
+                                             0);
```

Comment:
It might be better to draw the dot icon manually (using some font), because a black dot would be invisible on black background (i.e. when some themes are applied).

Reply:
That icon is a black dot surrounded by white border. No problem.

Reply:
Like this:
![like-this](https://user-images.githubusercontent.com/2107452/192194555-92ef9182-b5b7-4331-9496-8f432e041ee5.png)

Reply:
The bullet looks nice this way :)

Reply:
Heh I just checked, Windows does not recolor the bullet in "high contrast black" theme. So we do this better :) However maybe make it just a bit smaller? Looks quite bold.

Reply:
Fixed.
![black](https://user-images.githubusercontent.com/2107452/192392249-c8c71cef-4703-4877-acea-233352751e63.png)

![after](https://user-images.githubusercontent.com/2107452/192392250-77a1c816-c8a9-44ce-ba38-02d915385c6d.png)